plugins/discourse-assign/plugin.rb

```diff
@@ -892,19 +892,67 @@ module ::DiscourseAssign
   add_filter_custom_filter("assigned") do |scope, filter_values, guardian|
     next if !guardian.can_assign? || filter_values.blank?
 
-    user_or_group_name = filter_values.compact.first
+    # Handle multiple comma-separated values (user1,group1,user2)
+    names =
+      filter_values.compact.flat_map { |value| value.to_s.split(",") }.map(&:strip).reject(&:blank?)
 
-    next if user_or_group_name.blank?
+    next if names.blank?
 
-    if user_id = User.find_by_username(user_or_group_name)&.id
-      scope.where(<<~SQL, user_id)
-        topics.id IN (SELECT a.topic_id FROM assignments a WHERE a.assigned_to_id = ? AND a.assigned_to_type = 'User' AND a.active)
-      SQL
-    elsif group_id = Group.find_by(name: user_or_group_name)&.id
-      scope.where(<<~SQL, group_id)
-        topics.id IN (SELECT a.topic_id FROM assignments a WHERE a.assigned_to_id = ? AND a.assigned_to_type = 'Group' AND a.active)
-      SQL
+    if names.include?("nobody")
+      next scope.where("topics.id NOT IN (SELECT a.topic_id FROM assignments a WHERE a.active)")
+    end
+
+    if names.include?("*")
+      next scope.where("topics.id IN (SELECT a.topic_id FROM assignments a WHERE a.active)")
+    end
+
+    found_names, user_ids =
+      User.where(username_lower: names.map(&:downcase)).pluck(:username, :id).transpose
+
+    found_names ||= []
+    user_ids ||= []
+
+    # a bit edge casey cause we have username_lower for users but not for groups
+    # we share a namespace though so in practice this is ok
+    remaining_names = names - found_names
+    group_ids = []
+    group_ids.concat(Group.where(name: remaining_names).pluck(:id)) if remaining_names.present?
+
+    next scope.none if user_ids.empty? && group_ids.empty?
+
+    assignment_query = Assignment.none # needed cause we are adding .or later
+
+    if user_ids.present?
+      assignment_query =
+        assignment_query.or(
+          Assignment.active.where(assigned_to_type: "User", assigned_to_id: user_ids),
+        )
+    end
+
+    if group_ids.present?
+      assignment_query =
+        assignment_query.or(
+          Assignment.active.where(assigned_to_type: "Group", assigned_to_id: group_ids),
+        )
+    end
+
+    scope.where(id: assignment_query.select(:topic_id))
+  end
+
+  register_modifier(:topics_filter_options) do |results, guardian|
+    if guardian&.can_assign?
```

Comment:
```suggestion
    if guardian.can_assign?
```

Guardian is always present it seems